src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1847,13 +1847,25 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     // - N/A
     winrt::fire_and_forget TermControl::_TerminalCursorPositionChanged()
     {
+        bool expectedFalse{ false };
+        if (!_coroutineDispatchStateUpdateInProgress.compare_exchange_weak(expectedFalse, true))
+        {
+            // somebody's already in here.
+            return;
+        }
+
         if (_closing.load())
         {
             return;
         }
 
         auto weakThis{ get_weak() };
 
+        // Even with muffling, we dispatch this tens of times _per hundredth of a second_.
+        // Muffle it harder.
+        static constexpr auto CursorUpdateQuiesceTime{ std::chrono::milliseconds(100) };
+        co_await winrt::resume_after(CursorUpdateQuiesceTime);

```

Comment:
Sorry, can you explain in plain English why this works?

It takes me effort to discern that there's one of them stuck in here with the state sitting at TRUE above that will reject all the other updates even if this one is about to be serviced just to slow it down.

Looking for something like 

// If we're the lucky one who gets through, we'll still wait 100ms to clog the atomic above so we don't service the cursor update too fast. If we get through and finish processing the update quickly but similar requests are still beating down the door above in the atomic, we may still update the cursor way more than is visible to anyone's eye, which is a waste of effort.

Reply:
I'll just wholesale copy this comment, it's great. thanks